### PR TITLE
connectivity: Add echo-ingress-l7-via-hostport-with-encryption testcase

### PR DESCRIPTION
**Pull request by jschwinger233**
Add testcase to cover https://github.com/cilium/cilium/issues/32899.

The new testcase has been tested by https://github.com/cilium/cilium/pull/33691, two failing checks are ci-e2e-upgrade and mergeability.

### CODE REVIEW

**Review by mhofstetter — APPROVED** [review]
Thanks!

File: connectivity/builder/echo_ingress_l7.go [hunk 1]
```diff
@@ -16,26 +16,44 @@ var echoIngressL7HTTPPolicyYAML string
...
+		WithFeatureRequirements(
+			features.RequireEnabled(features.L7Proxy),
+			// Once https://github.com/cilium/cilium/issues/33168 is fixed, we
+			// can enable for IPsec too.
+			features.RequireMode(features.EncryptionPod, "wireguard"),
+			// Otherwise pod->hostport traffic will be policy
+			// denied on the ingress of dest node when
+			// routing=vxlan + kpr=1 + bpf_masq=1
+			features.RequireEnabled(features.EncryptionNode),
+		).
```
Comment (julianwiedmann) [inline connectivity/builder/echo_ingress_l7.go:55]: Thank you! Looking forward - I think we should also run this test without encryption enabled :). And hopefully find a way to remove the requirement on NodeEncryption ...

**Review by julianwiedmann — APPROVED** [review]

File: connectivity/builder/echo_ingress_l7.go (hunk 1 above)
Comment (julianwiedmann) [reply connectivity/builder/echo_ingress_l7.go:55]: @jschwinger233 ah but one thing we *do* need (and this is arguably a CLI release blocker) - this test should only run with cilium 1.16.0 or newer.

Comment (jschwinger233) [reply connectivity/builder/echo_ingress_l7.go:55]: Yeah, on it.

Comment (julianwiedmann) [reply connectivity/builder/echo_ingress_l7.go:55]: > Thank you! Looking forward - I think we should also run this test without encryption enabled :). And hopefully find a way to remove the requirement on NodeEncryption ...

Moved this to https://github.com/cilium/cilium-cli/issues/2758.

**Pull request merged**